core/src/main/java/org/elasticsearch/action/count/CountRequest.java

```diff
@@ -240,13 +192,6 @@ public String toString() {
     }
 
     public SearchRequest toSearchRequest() {
-        SearchRequest searchRequest = new SearchRequest(indices());
-        searchRequest.indicesOptions(indicesOptions());
-        searchRequest.types(types());
-        searchRequest.routing(routing());
-        searchRequest.preference(preference());
-        searchRequest.source(source());
-        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
         searchSourceBuilder.size(0);
```

Comment:
this one changes the state of the current SearchSourceBuilder. I wish there was a better way. INstead of silently setting the size to 0, shall we throw exception if it's != 0 ?